pyface/testing/event_loop_helper.py

```diff
@@ -0,0 +1,127 @@
+# (C) Copyright 2019 Enthought, Inc., Austin, TX
+# All right reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+
+
+import contextlib
+
+from traits.api import HasStrictTraits, Instance
+
+from pyface.gui import GUI
+from pyface.i_gui import IGUI
+from pyface.timer.api import CallbackTimer, EventTimer
+
+
+class ConditionTimeoutError(RuntimeError):
+    pass
+
+
+class EventLoopHelper(HasStrictTraits):
+    """ Toolkit-independent methods for running event loops in tests.
+    """
+
+    #: A reference to the GUI object
+    gui = Instance(IGUI, factory=GUI)
+
+    @contextlib.contextmanager
+    def dont_quit_when_last_window_closed(self):
+        """ Suppress exit of the application when the last window is closed.
+        """
+        flag = self.gui.quit_on_last_window_close
+        self.gui.quit_on_last_window_close = False
+        try:
+            yield
+        finally:
+            self.gui.quit_on_last_window_close = flag
+
+    def event_loop(self, repeat=1, allow_user_events=True):
```

Comment:
I'd really like to include some advice in the docstring that if you find yourself needing to use a `repeat` of greater than `1`, you should consider whether you can use `event_loop_until_condition` instead.